compiler/rustc_abi/src/layout.rs

```diff
@@ -357,10 +359,8 @@ pub trait LayoutCalculator {
                 // It'll fit, but we need to make some adjustments.
                 match layout.fields {
                     FieldsShape::Arbitrary { ref mut offsets, .. } => {
-                        for (j, offset) in offsets.iter_enumerated_mut() {
-                            if !variants[i][j].0.is_zst() {
```

Comment:
This is one of these places special-casing ZST that I removed

Reply:
This got added in https://github.com/rust-lang/rust/commit/d7a750b50436fbd228b176f6438566625e235990 by @mikebenfield. But their tests still pass after removing the `if`. So... maybe it just was never needed?